internal/services/appservice/helpers/app_stack.go

```diff
@@ -41,341 +42,369 @@ type ApplicationStackWindows struct {
-						"site_config.0.application_stack.0.php_version",
-						"site_config.0.application_stack.0.python",
-						"site_config.0.application_stack.0.python_version",
-					},
-				},
+	r := &pluginsdk.Resource{
+		Schema: map[string]*pluginsdk.Schema{
+			"dotnet_version": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{ // Note: DotNet versions are abstracted between API and Portal displayed values, so do not match 1:1. A table of the converted values is provided in the resource doc.
+					"v2.0",
+					"v3.0",
+					"v4.0",
+					"v5.0",
+					"v6.0",
+					"v7.0"}, false),
+				AtLeastOneOf: windowsApplicationStackConstraint,
+			},
 
-				"dotnet_core_version": {
-					Type:     pluginsdk.TypeString,
-					Optional: true,
-					ValidateFunc: validation.StringInSlice([]string{
-						"v4.0",
-					}, false),
-					AtLeastOneOf: []string{
-						"site_config.0.application_stack.0.docker_container_name",
-						"site_config.0.application_stack.0.dotnet_version",
-						"site_config.0.application_stack.0.dotnet_core_version",
-						"site_config.0.application_stack.0.java_version",
-						"site_config.0.application_stack.0.node_version",
-						"site_config.0.application_stack.0.php_version",
-						"site_config.0.application_stack.0.python",
-						"site_config.0.application_stack.0.python_version",
-					},
-					Description: "The version of DotNetCore to use.",
-				},
+			"dotnet_core_version": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"v4.0",
+				}, false),
+				AtLeastOneOf: windowsApplicationStackConstraint,
+				Description:  "The version of DotNetCore to use.",
+			},
 
-				"php_version": {
-					Type:     pluginsdk.TypeString,
-					Optional: true,
-					Computed: true,
-					ValidateFunc: validation.StringInSlice([]string{
-						PhpVersionSevenPointOne,  // Deprecated
-						PhpVersionSevenPointFour, // Deprecated
-						PhpVersionOff,            // Portal displays `Off` for `""` meaning use latest available
-					}, false),
-					AtLeastOneOf: []string{
-						"site_config.0.application_stack.0.docker_container_name",
-						"site_config.0.application_stack.0.dotnet_version",
-						"site_config.0.application_stack.0.dotnet_core_version",
-						"site_config.0.application_stack.0.java_version",
-						"site_config.0.application_stack.0.node_version",
-						"site_config.0.application_stack.0.php_version",
-						"site_config.0.application_stack.0.python",
-						"site_config.0.application_stack.0.python_version",
-					},
-				},
+			"php_version": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					PhpVersionSevenPointOne,  // Deprecated
+					PhpVersionSevenPointFour, // Deprecated
+					PhpVersionOff,            // Portal displays `Off` for `""` meaning use latest available
```

Comment:
in 4.x we should remove `Off` and `Computed: true`, meaning we can set this to an empty string when it's `Off` or `""`